apps/correlations/kinds/correlations.cue

```diff
@@ -0,0 +1,35 @@
+package kinds
+
+correlationsv0alpha1: {
+	kind:       "Correlation"  // note: must be uppercase
+	pluralName: "Correlations"
+	schema: {
+		spec: {
+			source_uid:  string
+			target_uid?:  string
+			label:       string
+			description?: string
+			config:      ConfigSpec
+			provisioned: bool
+			type:        CorrelationType
+		}
+	}
+}
+
```

Comment:
```suggestion
			source:  DataSourceRef
			target?:  DataSourceRef
			label:       string
			description?: string
			config:      ConfigSpec
			type:        CorrelationType
		}
	}
}

DataSourceRef: {
   group: string // grafana plugin id
   name: string  // grafana uid == k8s name
}
```

two changes:
1. avoid UID (it will confuse us forever 😢 -- and reference group + name)
2. lets support provisioning via annotations -- can ignore it for this PR, and we can set it when you implement legacy

Reply:
@ryantxu I have a few additional questions about this. I'm sorry for making you repeat information, these are just concepts I want to be sure we have clarified before moving forward and potentially making breaking changes.

1. The endpoint itself uses the datasource UID in the path - how do we plan on handling this if we want to move away from UIDs?
2. I just want to confirm the change of terminology here, as the correlation needs to be associated with one instance of a datasource, and not all instances belonging to one plugin. Is group referring to a UID or another identifier of one instance of a datasource, and name referring to the type, like prometheus or loki?